### PR TITLE
Update _getMsgType for URC without prefix

### PR DESCRIPTION
**Pull request by chinglee-iot**
Description
-----------
Update `_getMsgType` function for feedback from forum in [this post.](https://forums.freertos.org/t/getmsgtype-in-freertos-cellular-interface-library/18497)

* Update the logic for port not using URC without prefix table
* Check AT command type for response without prefix

Test Steps
-----------
N/A

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/cellular_pktio.c [hunk 1]
```diff
@@ -470,23 +472,34 @@ static _atRespType_t _getMsgType( CellularContext_t * pContext,
...
                 atRespType = AT_UNSOLICITED;
             }
         }
         else
         {
-            if( ( ( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND ) && ( pRespPrefix == NULL ) ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_DATA_WO_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_WITH_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE ) )
```
Comment (ActoryOu) [inline source/cellular_pktio.c:486]: Why do you replace these conditions with `CELLULAR_AT_NO_COMMAND` only?

**Review by ActoryOu — COMMENTED** [review]
Could you update the forum link on PR?

File: source/cellular_pktio.c (hunk 1 above)
Comment (chinglee-iot) [reply source/cellular_pktio.c:486]: The logic here is to handle token without prefix. The original implementation covers all the AT command type here.
```
typedef enum CellularATCommandType
{
    CELLULAR_AT_NO_RESULT,                  /**<  no response expected, only OK, ERROR etc. */
    CELLULAR_AT_WO_PREFIX,                  /**<  string response without a prefix. */
    CELLULAR_AT_WITH_PREFIX,                /**<  string response with a prefix. */
    CELLULAR_AT_MULTI_WITH_PREFIX,          /**<  multiple line response all start with a prefix. */
    CELLULAR_AT_MULTI_WO_PREFIX,            /**<  multiple line response with or without a prefix. */
    CELLULAR_AT_MULTI_DATA_WO_PREFIX,       /**<  multiple line data response with or without a prefix. */
    CELLULAR_AT_WO_PREFIX_NO_RESULT_CODE,   /**<  string response without a prefix and no result code is expected. */
    CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE, /**<  string response WITH a prefix and no result code is expected. */
    CELLULAR_AT_NO_COMMAND                  /**<  no command is waiting response. */
} CellularATCommandType_t;
```


```
( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND ) && ( pRespPrefix == NULL )
```
=>
```
    CELLULAR_AT_NO_RESULT,                  /**<  no response expected, only OK, ERROR etc. */
    CELLULAR_AT_WO_PREFIX,                  /**<  string response without a prefix. */
    CELLULAR_AT_WITH_PREFIX,                /**<  string response with a prefix. */
    CELLULAR_AT_WO_PREFIX_NO_RESULT_CODE,   /**<  string response without a prefix and no result code is expected. 
```

```
( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_DATA_WO_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_WITH_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE ) )
```
=>
```
    CELLULAR_AT_MULTI_WITH_PREFIX,          /**<  multiple line response all start with a prefix. */
    CELLULAR_AT_MULTI_WO_PREFIX,            /**<  multiple line response with or without a prefix. */
    CELLULAR_AT_MULTI_DATA_WO_PREFIX,       /**<  multiple line data response with or without a prefix. */
    CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE, /**<  string response with a prefix and no result code is expected. */
```

These conditions can be simplified with `( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND )`. The `pRespPrefix` will be checked in `_Cellular_PktioSendAtCmd`.

**Comment by chinglee-iot** [comment]
> Could you update the forum link on PR?

Updated in the description.

**Review by ActoryOu — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Pull request merged**